src/cargo/core/compiler/fingerprint/mod.rs

```diff
@@ -65,6 +65,7 @@
 //! Target Name                                | ✓           | ✓
 //! TargetKind (bin/lib/etc.)                  | ✓           | ✓
 //! Enabled Features                           | ✓           | ✓
+//! Declared Features                          | ✓           | ?
```

Comment:
FYI instead of just dropping a `?` and hoping reviewers pick it up, it helps if you call this kind of thing out directly.

You are only adding it to the Finterprint and not the Metadata, so this should be correct.  I think the choice of Fingerprint (overwrite same file on change) over Metadata (create unique file) is correct.